### PR TITLE
Partially implement MapboxNavigationTests in SPM

### PR DESCRIPTION
**Pull request by S2Ler**
- All assets moved to TestHelper target for easier maintenance.
- Introduced a new test helper target - "CarPlayTestHelper", which holds
some mocks required for testing CarPlay. It had to be written in ObjC
because Swift forbids specific unsafe patterns. In particular:
CPNavigationSession and CPInterfaceController init methods are marked
unavailable.
- Use https://github.com/alanzeino/ios-snapshot-test-case.git in SPM,
although, it is likely to be replaced with https://github.com/pointfreeco/swift-snapshot-testing
which doesn't have all the features but has first-class Xcode and SPM
support. Or extended with a custom fork to support more configuration
points.
- All snapshot-based tests are disabled because it is not clear how to
provide configuration for https://github.com/alanzeino/ios-snapshot-test-case.git
version with SPM.
- Xcode base MapboxNavigationTests are not tested and should be fixed
separately.

This is a first step. The next ones can be:
- Migrate to https://github.com/pointfreeco/swift-snapshot-testing or fork https://github.com/alanzeino/ios-snapshot-test-case.git with the needed functionality added. 
- Restore Xcode based testing as well.
- Think how to fix some tests.

Anyway, this PR will let you write new tests using SPM.

### CODE REVIEW

**Comment by S2Ler** [comment]
@1ec5 I need this PR to be reviewed as it blocks a lot of other PRs and work.

**Comment by S2Ler** [comment]
Follow up: mapbox/mapbox-navigation-ios#3043 builds upon this PR and migrates to a new snapshots testing library.

File: Package.swift
Comment (1ec5) [inline Package.swift:114]: Is the issue that SPM runs tests without a bundle that includes Info.plist? In #2876, @ShanMa1991 worked around that issue by having Mapbox(Core)Navigation look for a key in UserDefaults as an optional alternative to Info.plist; the user default would only be set inside the test. (#2911 implemented a more robust solution for Info.plist keys that are needed during production, but that probably isn’t necessary here.)

File: Package.swift
Comment (1ec5) [inline Package.swift:84]: Good idea to split out this target for the Objective-C-based CarPlay mocking hack.

File: MapboxNavigation.xcodeproj/project.pbxproj
Comment (1ec5) [inline MapboxNavigation.xcodeproj/project.pbxproj:2254]: If I understand correctly, this target and framework exists for running MapboxNavigationTests under SPM; do we even need it for running the tests under Carthage? The target seems to exist solely for some Objective-C code. It’s a similar story with CMapboxCoreNavigation and CTestHelper, but we simplified things a bit by keeping the targets and products unified under Carthage with an alternative umbrella header that SPM ignores.

File: Tests/MapboxNavigationTests/SKUTests.swift
Comment (1ec5) [inline Tests/MapboxNavigationTests/SKUTests.swift:27]: Is there an alternative way to present a view controller in this environment, or would we perhaps split out a separate XCUITest target (as tail work)?

File: Tests/MapboxNavigationTests/Support/AccessToken.swift
Comment (1ec5) [inline Tests/MapboxNavigationTests/Support/AccessToken.swift:17]: This works, though it’s possible that lower-level libraries may introduce more rigorous client-side validation of access tokens in the future. MapboxDirectionsTests uses [a cheeky placeholder access token](https://github.com/mapbox/mapbox-directions-swift/blob/60b4a028119cef727648d981d2daccea0641138c/Tests/MapboxDirectionsTests/DirectionsTests.swift#L14-L15) that might be easier to spot while debugging.

**Review by 1ec5 — APPROVED** [review]
I meant to hit approve. I don’t think my feedback in https://github.com/mapbox/mapbox-navigation-ios/pull/3029#pullrequestreview-673771531 is critical enough to block this PR.

File: Package.swift
Comment (S2Ler) [reply Package.swift:114]: I think I checked the stack trace and there wasn't a way to provide this access token at that time. I will get back to this after the chain of build system PRs will be merged.

File: Tests/MapboxNavigationTests/SKUTests.swift
Comment (S2Ler) [reply Tests/MapboxNavigationTests/SKUTests.swift:27]: We can't run XCUITests in SPM, can we? I think I need to put `#if SWIFT_PACKAGE` instead of disabling. 

Adding to the list of tail work. Will get back to this soon.

File: Tests/MapboxNavigationTests/Support/AccessToken.swift
Comment (S2Ler) [reply Tests/MapboxNavigationTests/Support/AccessToken.swift:17]: Thank you for your suggestion, I will change from "any" to "pk.feedCafeDadeDeadBeef-BadeBede.FadeCafeDadeDeed-BadeBede" and reuse "BogusToken" name convention.

File: MapboxNavigation.xcodeproj/project.pbxproj
Comment (S2Ler) [reply MapboxNavigation.xcodeproj/project.pbxproj:2254]: I added this target to the Xcode project so that the imports like https://github.com/mapbox/mapbox-navigation-ios/blob/270bf280af4edc3d5bcec0043d4e5b18f2f740a6/Tests/MapboxNavigationTests/CarPlayNavigationViewControllerTests.swift#L8 would work in both Carthage and SPM. I can wrap them in `#if SWIFT_PACKAGE`, but it seems to bring more burden for future changes. You must not forget to add this ifdef when writing code using the SPM package in Xcode.

File: Tests/MapboxNavigationTests/Support/AccessToken.swift
Comment (S2Ler) [reply Tests/MapboxNavigationTests/Support/AccessToken.swift]: Fixed in https://github.com/mapbox/mapbox-navigation-ios/pull/3029/commits/473c67c8dcd37037e0b72ff3311ddcc375a8007e

File: Package.swift
Comment (S2Ler) [reply Package.swift:114]: So here is what happens:

`CarPlayMapViewController.setupPassiveLocationManager()` creates `PassiveLocationDataSource` which in turn accesses `Directions.shared`. `Directions.shared` is initialized with the `DirectionsCredentials()` which looks for access token only in Info.plist.

Comment (1ec5) [reply Package.swift:114]: > `CarPlayMapViewController.setupPassiveLocationManager()` creates `PassiveLocationDataSource` which in turn accesses `Directions.shared`.

It’s potentially problematic that `Directions.shared` is used in this case. What if a developer needs to specify an access token programmatically rather than in Info.plist? Most other classes have a way to specify the credentials explicitly on initialization.

Comment (S2Ler) [reply Package.swift:114]: I'm happy to address this concern as a new issue.

**Pull request merged**
